config/filter.d/nginx-bad-request.conf

```diff
@@ -5,7 +5,7 @@
 
 # The request often doesn't contain a method, only some encoded garbage
 # This will also match requests that are entirely empty
-failregex = ^<HOST> - \S+ \[\] "[^"]*" 400
+failregex = ^<HOST> - \S+ \[\] "[^"]*" [4-5]0[0-9]
```

Comment:
The issue with that: strictly seen only 400 (and possibly 444) is bad request response code.
Someone would probably need to consider 403 (Forbidden) or 40[56] (Not allowed, Not acceptable) also as such "bad request", but definitely not 404, not to mention 401.

The issue with 404 is: it may be too heavy: few "broken" or outdated urls to some resources may cause a ban for fully legitimate users.
The issue with 401 is still worse, because it can be part of authentication handshake and such RE must additionally consider the user name (e. g. whether it is not empty).
Anyway we could accept something like this:
```diff
+ code = 400
- failregex = ^<HOST> - \S+ \[\] "[^"]*" 400
+ failregex = ^<HOST> - \S+ \[\] "[^"]*" <code>
```
so it'd be possible to set the code via filter in jail, e. g.:
```ini
filter = %(known/filter)s[code="(?!401)[45]\d{2}"]
```
but definitely not such code range by default... Sorry, no chances.